Spatial.gd

```diff
@@ -632,31 +638,18 @@ func gen_new_icon(position: Vector3, texture_path: String, waypoint_icon, catego
 	var category_data = category_item.get_metadata(0)
 	category_data.category3d.add_icon(new_icon)
 
-# This function take all of the currently rendered objects and converts it into
-# the data format that is saved/loaded from.
-func data_from_renderview():
-	var icons_data = []
-	var paths_data = []
-
-	for icon in $Icons.get_children():
-		icons_data.append({
-			"position": [icon.translation.x, icon.translation.y, -icon.translation.z],
-			"texture": icon.texture_path
-		})
+################################################################################
+# Section of functions for saving changes to markers
+################################################################################
+func on_change_made():
+	self.unsaved_data_icon.visible = true
 
-	for path in $Paths.get_children():
-		#print(path)
-		var points = []
-		for point in range(path.get_point_count()):
-			var point_position:Vector3 = path.get_point_position(point)
-			points.append([point_position.x, point_position.y, -point_position.z])
-		paths_data.append({
-			"points": points,
-			"texture": path.texture_path
-		})
-
-	var data_out = {"icons": icons_data, "paths": paths_data}
-	return data_out
+func save_current_map_data():
+	var packed_bytes = self.waypoint_data.to_bytes()
+	if packed_bytes.size() > 0:
```

Comment:
Does this check prevent saving a file which went from having "some markers" to "no markers" ?
IE: prevent deleting the last marker on a map?

Reply:
I was using the check to not make a blank file but you are correct it would prevent that. I'm going to remove the check.

Reply:
Make and link issue to track this theoretical case of creating blank files. Because that is possible, maybe the correct logic here will be to delete files that are empty instead of not saving/overwriting them. But something a future pass can deal with.